core/src/main/java/jenkins/util/SystemProperties.java

```diff
@@ -122,12 +124,15 @@ public void contextDestroyed(ServletContextEvent event) {
 
     /**
      * Mark a key whose value should be made accessible in agent JVMs.
+     *
+     * @param key Property key to be explicitly allowed
      */
     public static void allowOnAgent(String key) {
         ALLOW_ON_AGENT.add(key);
     }
 
     @Extension
+    @Restricted(NoExternalUse.class)
```

Comment:
BTW these two could probably even be `DoNotUse` (not sure offhand).

Reply:
Can be changed later